src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -573,6 +579,32 @@ public Void caseProxyObjectType(ProxyObjectType type) {
       });
     }
 
+    // This method is used for objects like the static props on an interface, that
+    // behave like an object with bunch of props, but are not technically defined as
+    // record types.
+    private void visitRecordLikeType(ObjectType type) {
+      emit("{");
+      Iterator<String> it = type.getOwnPropertyNames().iterator();
+      Boolean isFirst = true;
+      while (it.hasNext()) {
+        String propName = it.next();
+
+        // The prototype key is implicitly defined in the Object signature.
+        if (propName.equals("prototype")) continue;
+
+        if (isFirst) {
```

Comment:
I've used http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/PeekingIterator.html for this before - I think it's already available in our classpath? See if that makes it less awkward